elm/src/Test/Runner/Node.elm

```diff
@@ -309,11 +309,49 @@ init { processes, globs, paths, fuzzRuns, initialSeed, report, runners } _ =
     ( model, Cmd.none )
 
 
+{-| The implementation of this function will be replaced in the generated JS
+with a version that returns `Just value` if `value` is a `Test`.
+-}
+check : a -> Maybe Test
+check value =
+    Nothing
+
+
 {-| Run the tests.
 -}
-run : RunnerOptions -> Test -> Program Int Model Msg
-run { runs, seed, report, globs, paths, processes } test =
+run : RunnerOptions -> List ( String, List (Maybe Test) ) -> Program Int Model Msg
+run { runs, seed, report, globs, paths, processes } possiblyTests =
     let
+        tests =
+            possiblyTests
+                |> List.filterMap
+                    (\( moduleName, maybeModuleTests ) ->
+                        let
+                            moduleTests =
+                                List.filterMap identity maybeModuleTests
+                        in
+                        if List.isEmpty moduleTests then
+                            Nothing
+
+                        else
+                            Just (Test.describe moduleName moduleTests)
+                    )
+
+        test =
+            if List.isEmpty tests then
+                Test.todo
+                    (if List.isEmpty paths then
+                        "I couldn't find any exposed values of type Test in any *.elm files in the tests/ directory of your project's root directory.\n\nTo generate some initial tests to get things going, run elm-test init"
+
+                     else
+                        "I couldn't find any exposed values of type Test in files matching:\n"
+                            ++ String.join "\n" paths
+                            ++ "\n\nMaybe try running elm-test with no arguments?"
+                    )
+
```

Comment:
This check has moved from JS to Elm. I used `Test.todo` to get going with this experiment, but it feels a bit like a hack and I’m not sure if it gives the best output:

```
elm-test 0.19.1-revision4
-------------------------

Running 1 test. To reproduce these results, run: elm-test --fuzz 100 --seed 343653604167020 src


TEST RUN INCOMPLETE because there is 1 TODO remaining

Duration: 170 ms
Passed:   0
Failed:   0
Todo:     1
◦ TODO: I couldn't find any exposed values of type Test in files matching:
/Users/lydell/forks/node-test-runner/example-application/src/Something.elm

Maybe try running elm-test with no arguments?
```

Reply:
@harrysarson This is one of the few things left to do. Do we think the above message is good enough, or do you have ideas on how to handle this case better?

Reply:
I figured it out!

### Before I figured it out:

If the CLI arguments match no files:

```
❯ ../bin/elm-test 'src/**/*Test.elm'
No tests found for the file pattern "src/**/*Test.elm"

Maybe try running elm-test with no arguments?
```

If the CLI arguments do match files, but they contain no exposed Test values:

```
❯ ../bin/elm-test src

elm-test 0.19.1-revision4
-------------------------

Running 1 test. To reproduce these results, run: elm-test --fuzz 100 --seed 76422859129210 src


TEST RUN INCOMPLETE because there is 1 TODO remaining

Duration: 164 ms
Passed:   0
Failed:   0
Todo:     1
◦ TODO: I couldn't find any exposed values of type Test in files matching:
/Users/lydell/forks/node-test-runner/example-application/src/Something.elm

Maybe try running elm-test with no arguments?
```

### After

If the CLI arguments match no files:

```
❯ ../bin/elm-test 'src/**/*Test.elm'
No files found matching:

src/**/*Test.elm

Are the above patterns correct? Maybe try running elm-test with no arguments?
```

If the CLI arguments do match files, but they contain no exposed Test values:

```
❯ ../bin/elm-test src
No exposed values of type Test found in files matching:

src

Are the above patterns correct? Maybe try running elm-test with no arguments?

Are there tests in any of the matched files?
If not – add some!
If there are – are they exposed?
```